### PR TITLE
Rework attribution options to be specified by container authors

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

This reworks how options affecting attribution get specified--previous method relied heavily on the host setting things up appropriately. Some changes which I think are in the right direction:

1. "enabling attribution" needs to be done both at the container-runtime level and at the DDS level.
2. I've removed config flags--shying away from these seems reasonable as consumers can always inject things like feature flags of their own if they want quicker rollout

Notably, having individualized control at the DDS level controlled by a dataObject's creator seems like the right direction for the future: it's easy to imagine containers which have heterogeneous content involving some SharedStrings that want attribution enabled and other SharedStrings which don't. Making this setting parameterizable via the SharedObject factory satisfies both of those goals. One can even accomplish having the same type of DDS within the same data store with different parameters if enough information about those parameters is persisted in the attributes blob (though that isn't done here, this would require changes to the channel creation flow).

It's probably not ideal that the container author also needs to enable it at the runtime level, but not a huge deal. I also don't feel strongly that that setting *should* be injected via scope, but since this was already the case for other content, it didn't feel unreasonable.

### CODE REVIEW

**Comment by Abe27342** [comment]
There are still some things to resolve with this before it's in a check-in-able state:

1. I need to do a full audit of attribution docs/readmes to make sure they're up-to-date.
2. SharedCell needs to be updated
3. Unrelated to attribution, our testing setup (and maybe other things which are unknown) needs updates to make 'parameterizable DDSes' make sense. The main issue currently is that `describeCompat` and variants build a registry of all DDSes using the parameterless `getFactory` and take the subset of that registry which is referenced by a given test. See note on attribution e2e tests. this creates problems.

(3) is obviously the biggest thing here.

File: packages/dds/sequence/src/sequence.ts [hunk 1]
```diff
@@ -109,6 +110,10 @@ export interface ISharedSegmentSequenceEvents extends ISharedObjectEvents {
 	);
 }
 
+export interface SequenceOptions {
+	attribution: IMergeTreeAttributionOptions;
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/sequence.ts:114]: i wonder if we want to take a firmer stance on what options are, specifically if they can be more than primitive values. probably doesn't need to be solved immediately.

File: packages/dds/sequence/src/sequenceFactory.ts [hunk 2]
```diff
@@ -24,6 +32,8 @@ export class SharedStringFactory implements IChannelFactory {
 		packageVersion: pkgVersion,
 	};
 
+	public constructor(public readonly options?: SequenceOptions) {}
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/dds/sequence/src/sequenceFactory.ts:35]: we follow a similar pattern for data store and dds registries, and it works ok.

File: api-report/merge-tree.api.md [hunk 3]
```diff
@@ -343,6 +340,22 @@ export interface IAttributionCollectionSpec<T> {
     }>;
 }
 
+// @public
+export interface IAttributionPolicyFactory {
```
Comment (Abe27342) [inline api-report/merge-tree.api.md:344]: still need to do a bit of cleanup of API tags and docs in this realm.

Also not totally convinced I want to expose classes for the policy registry elements or have them be self-named, but that's a bit more minor (self-naming is less of a concern for closed-ecosystem set of options, which is what attribution policies currently are. Maybe we open them up in the future but it's pretty easy to shoot yourself in the foot, so I'd rather not if we can get away with it)

**Pull request closed without merging**